src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int INCREAS_COEFF = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int currentCapacity;
+    private float thrashLoad;
+    private Node<K, V>[] buckets;
+
+    {
+        buckets = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        currentCapacity = INITIAL_CAPACITY;
+        thrashLoad = currentCapacity * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= thrashLoad) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> bucketNode = findNode(key);
+        if (bucketNode == null) {
+            buckets[hash(key)] = newNode;
+            size++;
+            return;
+        }
+        if (key == bucketNode.key || (bucketNode.key != null && bucketNode.key.equals(key))) {
+            bucketNode.value = value;
+            return;
+        }
+        bucketNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> bucketNode = buckets[hash(key)];
+        while (bucketNode != null) {
+            if (key == bucketNode.key || (bucketNode.key != null && bucketNode.key.equals(key))) {
```

Comment:
```suggestion
            if (key == bucketNode.key || bucketNode.key != null && bucketNode.key.equals(key)) {
```
&& have higher [priority](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html)